skimage/util/_montage.py

```diff
@@ -0,0 +1,252 @@
+        If True, the last `arr_in` dimension is threated as a color channel,
+        otherwise as spatial.
+
+    Returns
+    -------
+    arr_out : (K*M+p, K*N+p[, C]) ndarray
+        Output array with input images glued together (including padding `p`).
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage.util.montage import montage
+    >>> arr_in = np.arange(3 * 2 * 2).reshape(3, 2, 2)
+    >>> arr_in  # doctest: +NORMALIZE_WHITESPACE
+    array([[[ 0,  1],
+            [ 2,  3]],
+           [[ 4,  5],
+            [ 6,  7]],
+           [[ 8,  9],
+            [10, 11]]])
+    >>> arr_out = montage(arr_in)
+    >>> arr_out.shape
+    (4, 4)
+    >>> arr_out
+    array([[ 0,  1,  4,  5],
+           [ 2,  3,  6,  7],
+           [ 8,  9,  5,  5],
+           [10, 11,  5,  5]])
+    >>> arr_in.mean()
+    5.5
+    >>> arr_out_nonsquare = montage(arr_in, grid_shape=(1, 3))
+    >>> arr_out_nonsquare
+    array([[ 0,  1,  4,  5,  8,  9],
+           [ 2,  3,  6,  7, 10, 11]])
+    >>> arr_out_nonsquare.shape
+    (2, 6)
+
+    """
+    if multichannel:
+        arr_in = np.asarray(arr_in)
+    else:
+        arr_in = np.asarray(arr_in)[..., np.newaxis]
+
+    if arr_in.ndim != 4:
+        raise ValueError('Input array has to be either 3- or 4-dimensional')
+
+    n_images, n_rows, n_cols, n_chan = arr_in.shape
+
+    if grid_shape:
+        ntiles_row, ntiles_col = map(int, grid_shape)
+    else:
+        ntiles_row = ntiles_col = int(np.ceil(np.sqrt(n_images)))
+
+    # Rescale intensity if necessary
+    if rescale_intensity:
+        for i in range(n_images):
+            arr_in[i] = exposure.rescale_intensity(arr_in[i])
+
+    # Calculate the fill value
+    if fill == 'mean':
+        fill = arr_in.mean(axis=(0, 1, 2))
+    fill *= np.ones(1)
+    fill = np.asarray(fill, dtype=arr_in.dtype)
+
+    # Pre-allocate an array with padding for montage
+    n_pad = padding_width
+    arr_out = np.empty(((n_rows+n_pad)*ntiles_row+n_pad,
+                        (n_cols+n_pad)*ntiles_col+n_pad,
+                        n_chan), dtype=arr_in.dtype)
+    for idx_chan in range(n_chan):
+        arr_out[..., idx_chan] = fill[idx_chan]
+
+    slices_row = [slice(n_pad+(n_rows+n_pad)*n,
+                        n_pad+(n_rows+n_pad)*n+n_rows)
+                  for n in range(ntiles_row)]
+    slices_col = [slice(n_pad+(n_cols+n_pad)*n,
+                        n_pad+(n_cols+n_pad)*n+n_cols)
+                  for n in range(ntiles_col)]
+
+    # Copy the data to the output array
+    for idx_image, image in enumerate(arr_in):
+        idx_sr = idx_image // ntiles_col
+        idx_sc = idx_image % ntiles_col
+        arr_out[slices_row[idx_sr], slices_col[idx_sc], :] = image
+
+    if multichannel:
+        return arr_out
+    else:
+        return arr_out[..., 0]
+
+
+@deprecated('montage', removed_version='0.15')
+def montage2d(arr_in, fill='mean', rescale_intensity=False, grid_shape=None,
+              padding_width=0):
```

Comment:
I just wanted to keep some backward compatibility. There is a slight difference between `montage` and `montage2d` in the way the padding is used - `montage2d` doesn't pad upper and left borders.